### PR TITLE
Issue #360 fix teams controller

### PR DESCRIPTION
**Pull request by YuriiChornyi**
### Summary

_Describe main changes going into those pull requests._

_Use "closes #(issue number)" to enable GitHub to close issue when PR is merged_

### Areas Of Interest

_Any specific places you want to bring attention of the reviewers? Some places you are uncertain in/some design decisions. (If none remove this section)_

#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references
- [ ] When new property is added all related comparers are updated

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs [hunk 1]
```diff
@@ -181,49 +181,32 @@ public JsonResult Edit(TeamViewModel teamViewModel)
...
-                    //var playersStillSame = playersInTeamDb.SequenceEqual(playersInTeamViewModelWhichHaveId);
-                    bool playersStillSame = !playersInTeamDb.Except(playersInTeamViewModelWhichHaveId).Any()
-                                            && !playersInTeamViewModelWhichHaveId.Except(playersInTeamDb).Any()
-                                            && playersInTeamDb.Count() == playersInTeamViewModelWhichHaveId.Count() 
-                                            && playersInTeamDb.Intersect(playersInTeamViewModelWhichHaveId).Count() 
-                                            == playersInTeamViewModelWhichHaveId.Count();
+
+                    CheckChangeCaptain(teamViewModel,team);
+
+                    var playersStillSame = !playersInTeamDb.Except(playersInTeamViewModelWhichHaveId).Any()
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: You don't need this.

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs [hunk 2]
```diff
@@ -369,19 +352,34 @@ private string PhotoPath(int id)
             return _fileService.FileExists(HttpContext.Request.MapPath(photoPath)) ? photoPath : string.Format(Constants.TEAM_PHOTO_PATH, 0);
         }
 
+        private void CheckChangeCaptain(TeamViewModel teamViewModel,Team team)
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: Why can't you issue ChangeCaptain call only if user edited this field?

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs [hunk 3]
```diff
@@ -177,53 +177,31 @@ public JsonResult Edit(TeamViewModel teamViewModel)
...
-                            .ToList())
-                            .Select(x => new PlayerId(x.Id))
-                            .ToList();
-                    }
-                    if (!playersStillSame)
-                    {
-                        playersIdToAddToTeam.AddRange(UpdateRoaster(teamViewModel.Roster, team, playersInTeamDb));
+                                .Where(x => x.Id == 0)
+                                .Select(x => x.ToCreatePlayerDto())
+                                .ToList())
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: Why to call TOList twice?

File: src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs [hunk 4]
```diff
@@ -123,10 +123,24 @@ public void AddPlayers(TeamId team, IEnumerable<PlayerId> players)
...
             changedTeam.AddPlayers(players);
             _teamRepository.Update(changedTeam);
         }
 
+        private void CheckIfPlayersAreNotInAnoutherTeam(IEnumerable<PlayerId> addedPlayers)
+        {
+            var playersTeams = addedPlayers
+                .Select(x => _getPlayerTeamQuery.Execute(new FindByPlayerCriteria { Id = x.Id }));
+            //check if players play in another team
+            if (playersTeams.Any(x => x > 0))
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs]: what does it mean x>0?

File: src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs [hunk 5]
```diff
@@ -154,7 +168,12 @@ public void Edit(Team teamToEdit)
         {
             _authService.CheckAccess(AuthOperations.Teams.Edit);
 
-            ThrowExceptionIfTeamWithSuchNameExists(teamToEdit.Name);
+            var teamInDb = _getTeamByNameQuery.Execute(new FindByNameCriteria { Name = teamToEdit.Name });
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs]: Make sure it's covered by tests

File: src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs (hunk 4 above)
Comment (oleggerus) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs]: It checks if any player is a member of any team

Comment (oleggerus) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs]: if no 'x' will be equals to null

Comment (Myroslavv) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TeamService.cs]: _getPlayerTeamQuery returns id of team in form of **int**, in which players is a member. If there is no such a team -  '0' is returned.

File: src/VolleyManagement.Backend/VolleyManagement.UI/Scripts/VMScripts/TeamOperations/CreateEdit.js [hunk 6]
```diff
@@ -254,6 +261,17 @@
     return result;
   };
 
+  function checkIfPlayerExists(playersWhichWasInTeam, playerId) {
+    var ifexist = false;
+    for (var i = 0; i < playersWhichWasInTeam.length; i++) {
+      if (+playersWhichWasInTeam[i].id === playerId) {
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Scripts/VMScripts/TeamOperations/CreateEdit.js]: what do you need this + at the beggining of the condition for?

Comment (YuriiChornyi) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Scripts/VMScripts/TeamOperations/CreateEdit.js]: https://medium.com/@nikjohn/cast-to-number-in-javascript-using-the-unary-operator-f4ca67c792ce

**Pull request merged**